cogs/kick_ghosts.py

```diff
@@ -0,0 +1,291 @@
+import asyncio
+from collections import namedtuple
+from datetime import datetime, timedelta
+from typing import Union
+
+from discord import Member, User, Role, Embed, Guild, TextChannel, Forbidden
+from discord.ext.commands import Cog, Bot, group, bot_has_guild_permissions, Context, BadArgument
+# noinspection PyPackageRequirements
+from discord.ext.tasks import loop
+from pymongo.errors import ServerSelectionTimeoutError
+
+from cogs.botStatus import listener
+from core.error.error_reply import startup_error_reply
+from core.globalEnum import CollectionEnum, ConfigurationNameEnum
+from core.logger import get_discord_child_logger
+from mongo.primitiveMongoData import PrimitiveMongoData
+
+logger = get_discord_child_logger("KickGhosts")
+event = namedtuple("KickTime", ["hour", "min"])
+
+
+class KickGhosts(Cog):
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.db: PrimitiveMongoData = PrimitiveMongoData(CollectionEnum.KICK_GHOSTS)
+        self.startup = True
+
+        # The actual values will be set at the on_ready method.
+        self.config: dict[ConfigurationNameEnum: Union[bool, int, event]] = {
+            ConfigurationNameEnum.ENABLED: False,
+            ConfigurationNameEnum.DEADLINE: 30,
+            ConfigurationNameEnum.WARNING: 7,
+            ConfigurationNameEnum.TIME: event(hour=8, min=0)
+        }
+
+        self.kick_not_verified.start()
+
+    def cog_unload(self):
+        self.kick_not_verified.stop()
+
+    @listener()
+    async def on_ready(self):
+        if self.startup:
+            self.startup = False
+            try:
+                for c in self.config:
+                    key = c.value
+                    default = self.config[c]
+
+                    enabled = await self.db.find_one({key: {"$exists": True}})
+                    if enabled:
+                        self.config[c] = enabled[key]
+                    else:
+                        await self.db.insert_one({key: default})
```

Comment:
You don't need the variable `default` since it is only used there

Reply:
Agree.